### PR TITLE
TF2 msgs not implemented in ros2 branch (#116)

### PR DESCRIPTION
**Pull request by Ubb90**
* add ign pose_v to ros tf2_message bridge

* add tf2 msgs dependency

### CODE REVIEW

**Comment by Ubb90** [comment]
I'll fix the test tomorrow. Not too sure why the tf2 test doesn't pass as ignition correctly register and echo the topic fine in terminal.

File: ros_ign_bridge/README.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ service calls. Its support is limited to only the following message types:
 | sensor_msgs/LaserScan          | ignition::msgs::LaserScan        |
 | sensor_msgs/MagneticField      | ignition::msgs::Magnetometer     |
 | sensor_msgs/PointCloud2        | ignition::msgs::PointCloudPacked |
+| tf_msgs/TFMessage              | ignition::msgs::Pose_V           |
```
Comment (chapulina) [inline ros_ign_bridge/README.md]:
```suggestion
| tf2_msgs/TFMessage              | ignition::msgs::Pose_V           |
```

File: ros_ign_bridge/package.xml [hunk 2]
```diff
@@ -19,6 +19,8 @@
   <depend>rosgraph_msgs</depend>
   <depend>sensor_msgs</depend>
   <depend>std_msgs</depend>
+  <depend>std_srvs</depend>
```
Comment (chapulina) [inline ros_ign_bridge/package.xml]: Is `std_srvs` needed?

File: ros_ign_bridge/test/test_utils.hpp [hunk 3]
```diff
@@ -888,6 +905,21 @@ void compareTestMsg(const std::shared_ptr<ignition::msgs::Float> & _msg)
   EXPECT_FLOAT_EQ(expected_msg.data(), _msg->data());
 }
 
+void createTestMsg(ignition::msgs::Double & _msg)
```
Comment (chapulina) [inline ros_ign_bridge/test/test_utils.hpp:930]: Can you add a documentation `\brief`?

File: ros_ign_bridge/test/test_utils.hpp [hunk 4]
```diff
@@ -888,6 +905,21 @@ void compareTestMsg(const std::shared_ptr<ignition::msgs::Float> & _msg)
...
+}
+
+/// \brief Compare a message with the populated for testing.
+/// \param[in] _msg The message to compare.
+void compareTestMsg(const std::shared_ptr<ignition::msgs::Double> & _msg)
+{
+    ignition::msgs::Double expected_msg;
+    createTestMsg(expected_msg);
+
+    EXPECT_DOUBLE_EQ(expected_msg.data(), _msg->data());
```
Comment (chapulina) [inline ros_ign_bridge/test/test_utils.hpp]: Please use 2-spaces indentation

File: ros_ign_bridge/test/test_utils.hpp [hunk 5]
```diff
@@ -395,6 +395,23 @@ void compareTestMsg(const std::shared_ptr<geometry_msgs::msg::TransformStamped>
   EXPECT_EQ(expected_msg.child_frame_id, _msg->child_frame_id);
 }
 
+void createTestMsg(tf2_msgs::msg::TFMessage &_msg)
```
Comment (chapulina) [inline ros_ign_bridge/test/test_utils.hpp]: Please document this function

File: ros_ign_bridge/test/subscribers/ros_subscriber.cpp [hunk 6]
```diff
@@ -239,6 +239,18 @@ TEST(ROSSubscriberTest, TransformStamped)
   EXPECT_TRUE(client.callbackExecuted);
 }
 
+/////////////////////////////////////////////////
+TEST(ROSSubscriberTest, TF2Message)
```
Comment (chapulina) [inline ros_ign_bridge/test/subscribers/ros_subscriber.cpp:255]: Did you mean to also add the "double" tests? You've added support for double in a few places.

**Review by chapulina — CHANGES_REQUESTED** [review]
> I'll fix the test tomorrow.

Thanks! There are some uncrustify (style) errors and also some test failures. I also think you should either remove the support for `double` or complete the tests.

It looks like you're missing some lines on the launch files:

https://github.com/ignitionrobotics/ros_ign/tree/ros2/ros_ign_bridge/test/launch

File: ros_ign_bridge/test/subscribers/ros_subscriber.cpp (hunk 6 above)
Comment (Ubb90) [reply ros_ign_bridge/test/subscribers/ros_subscriber.cpp:255]: yes but I forgot to add the test and most of the implementation as well. My bad sorry. It should be fixed now

File: ros_ign_bridge/package.xml (hunk 2 above)
Comment (Ubb90) [reply ros_ign_bridge/package.xml]: It was added in 7b06c21 but it wasn't needed for just that so I removed it. Good call

**Comment by Ubb90** [comment]
I updated the Readme too because the list was not reflecting the correct node name. If it should be in a separate issue/PR tell me and I'll revert it

File: ros_ign_bridge/src/factories.cpp [hunk 7]
```diff
@@ -167,7 +177,19 @@ get_factory_impl(
...
+  if (
+    (ros_type_name == "tf2_msgs/msg/TFMessage" || ros_type_name == "") &&
+    ign_type_name == "ignition.msgs.Pose_V")
+  {
+    return std::make_shared<
+      Factory<
+        tf2_msgs::msg::TFMessage,
+        ignition::msgs::Pose_V
+      >
+    >("tf2_msgs/TFMessage", ign_type_name);
```
Comment (chapulina) [inline ros_ign_bridge/src/factories.cpp]:
```suggestion
    >("tf2_msgs/msg/TFMessage", ign_type_name);
```

File: ros_ign_bridge/test/subscribers/ign_subscriber.cpp [hunk 8]
```diff
@@ -211,6 +211,18 @@ TEST(IgnSubscriberTest, TransformStamped)
   EXPECT_TRUE(client.callbackExecuted);
 }
```
Comment (chapulina) [inline ros_ign_bridge/test/subscribers/ign_subscriber.cpp:225]: Should we add a double test here?

**Review by chapulina — COMMENTED** [review]
CI is happy now, nice work! Just a couple more comments.

File: ros_ign_bridge/test/subscribers/ign_subscriber.cpp (hunk 8 above)
Comment (Ubb90) [reply ros_ign_bridge/test/subscribers/ign_subscriber.cpp:225]: I added one but there was none present in the melodic branch so maybe one should be added there too.

Comment (chapulina) [reply ros_ign_bridge/test/subscribers/ign_subscriber.cpp:225]: Thanks! And good catch. Yeah ideally we'd have one there too.

**Review by chapulina — APPROVED** [review]

**Pull request merged**